config/routes/internal_api.rb

```diff
@@ -35,5 +36,10 @@
     namespace :payments do
       resources :providers, only: [:create, :index, :update]
     end
+
+    resource :profile, only: [:update], controller: "profile" do
+      delete "/remove_avatar", to: "profile#remove_avatar"
+      get "/", to: "profile#index"
+    end
```

Comment:
```suggestion
    resource :profile, only: [:update, :show], controller: "profile" do
      delete "/remove_avatar", to: "profile#remove_avatar"
    end
```
Use show action instead of index in singular route